### PR TITLE
Added create methods for domain, cdn and waf + entities

### PR DESCRIPTION
**Pull request by BenjiHartley**

### CODE REVIEW

File: src/DDoSX/DomainClient.php [hunk 1]
```diff
@@ -47,6 +49,22 @@ public function getByName($domainName)
         return $this->serializeDomain($body->data);
     }
 
+    /**
+     * @param Domain $domain
+     * @return SelfResponse
+     */
+    public function create(Domain $domain)
+    {
+        $response = $this->post('v1/domains', json_encode($this->friendlyToApi($domain, $this->requestMap)));
```
Comment (martinshaw) [inline src/DDoSX/DomainClient.php]: Where is `requestMap` declared in this client class? It is declared in the admin class

Comment (martinshaw) [inline src/DDoSX/DomainClient.php]: Where is requestMap declared in this client class? It is declared in the admin class

File: src/SelfResponse.php [hunk 2]
```diff
@@ -27,10 +27,11 @@ class SelfResponse
     /**
      * SelfResponse constructor.
      * @param $response
+     * @param string $idProperty
      */
-    public function __construct($response)
+    public function __construct($response, $idProperty = "id")
```
Comment (martinshaw) [inline src/SelfResponse.php]: Nit: using single quotes when we do not need string interpolation will give slight performance improvement.

File: src/DDoSX/CdnClient.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $response = $this->post(
+            'v1/domains/' . $cdn->name . '/cdn',
+            json_encode($this->friendlyToApi($cdn, $this->requestMap))
+        );
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return (new SelfResponse($body, "domain_name"))
+            ->setClient($this)
+            ->serializeWith(function ($response) {
+                return new Cdn($this->apiToFriendly($response->data[0], $this->requestMap));
```
Comment (martinshaw) [inline src/DDoSX/CdnClient.php]: Where is requestMap declared in this client class? It is declared in the admin class

File: src/DDoSX/DomainClient.php (hunk 1 above)
Comment (BenjiHartley) [reply src/DDoSX/DomainClient.php]: updated and added the requestMap here

File: src/DDoSX/CdnClient.php (hunk 3 above)
Comment (BenjiHartley) [reply src/DDoSX/CdnClient.php]: Updated it :+1:

File: src/SelfResponse.php (hunk 2 above)
Comment (BenjiHartley) [reply src/SelfResponse.php]: Updated it :+1:

**Comment by phily245** [comment]
LGTM

**Pull request merged**